support/k8s/k8s-workload-registrar/config_crd.go

```diff
@@ -51,6 +55,41 @@ func (c *CRDMode) ParseConfig(hclConfig string) error {
 		c.WebhookPort = defaultWebhookPort
 	}
 
+	if c.IdentityTemplate != "" && c.CommonMode.PodAnnotation != "" {
+		return errs.New("workload registration configuration is incorrect, can't specify both identity_template and pod_annotation")
+	}
+
+	if c.IdentityTemplate != "" && c.CommonMode.PodLabel != "" {
+		return errs.New("workload registration configuration is incorrect, can't specify both identity_template and pod_label")
+	}
+
+	// eliminate orphaned identity_template_label
+	if c.IdentityTemplateLabel != "" && c.IdentityTemplate == "" {
+		return errs.New("identity_template_label defined without identity_template")
+	}
+
+	// eliminate orphaned context
+	if c.Context != nil && c.IdentityTemplate == "" {
+		return errs.New("context defined without identity_template")
+	}
+
+	// eliminate reference to non-existing context
+	if c.Context == nil && c.IdentityTemplate != "" && strings.Contains(c.IdentityTemplate, "{{.Context.") {
+		return errs.New("identity_template references non-existing context")
+	}
+
+	// IdentityTemplate represents the format following the trust domain and as such, it must not begin with spiffe://, // or /
+	if strings.HasPrefix(c.IdentityTemplate, "spiffe://") ||
+		strings.HasPrefix(c.IdentityTemplate, "/") {
+		return errs.New("identity_template cannot start with spiffe:// or /")
+	}
+
+	if c.CommonMode.Mode == modeCRD {
+		if c.IdentityTemplate == "" && c.CommonMode.PodAnnotation == "" && c.CommonMode.PodLabel == "" {
+			return errs.New("in crd mode, only one workload registration mode is allowed: identity_template, pod_annotation or pod_label")
```

Comment:
this seems to want to check the existence of 2 or more, the conditional doesn't seem to do that, it only checks for existence of all 3